### PR TITLE
qmk: new port

### PR DESCRIPTION
**Pull request by bashu**
#### Description

A program to help users work with [QMK Firmware](https://qmk.fm/).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.2 21G320 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by bashu** [comment]
Depends on https://github.com/macports/macports-ports/pull/17156

**Comment by bashu** [comment]
cc @reneeotten

File: sysutils/qmk/Portfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+long_description    {*}${description}
+
+homepage            https://github.com/qmk/qmk_cli
+
+checksums           rmd160  196cb96b9cdc85b705eb0115aa034d9da857f6bb \
+                    sha256  dd028e09ebcd61f8bdf8cb82929dfafc0e007d97a5a3803b45819b4641773269 \
+                    size    11994
+
+python.default_version \
+    310
```
Comment (reneeotten) [inline sysutils/qmk/Portfile:28]: Python 3.21 is the current default in MacPorts. The port should be updated to use this if possible.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: sysutils/qmk/Portfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    port:dfu-util \
+    port:py${python.version}-appdirs \
+    port:py${python.version}-argcomplete \
+    port:py${python.version}-colorama \
+    port:py${python.version}-dotty-dict \
+    port:py${python.version}-hid \
+    port:py${python.version}-hjson \
+    port:py${python.version}-jsonschema \
+    port:py${python.version}-milc \
+    port:py${python.version}-pillow \
```
Comment (reneeotten) [inline sysutils/qmk/Portfile:48]: pillow should be capitalized as that is the portname; otherwise it will fail on case-sensitive systems

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
@bashu can you please make the requested changes so that we can merge this PR?

File: sysutils/qmk/Portfile (hunk 1 above)
Comment (bashu) [reply sysutils/qmk/Portfile]: change to 3.11

File: sysutils/qmk/Portfile (hunk 2 above)
Comment (bashu) [reply sysutils/qmk/Portfile]: capitalized as requested

**Comment by bashu** [comment]
to @reneeotten please review

**Comment by reneeotten** [comment]
@bashu what's the timeframe for getting this PR ready? Is there anything we can do to help?

**Comment by reneeotten** [comment]
No response in 2 months... closing this PR. Feel free to reopen once you found the time to finalize this.

**Pull request closed without merging**